### PR TITLE
feat(components): [form] add form item relations validation

### PR DESCRIPTION
**Pull request by guozi9999**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### Description

Add a new `relations` prop to `ElFormItem` to revalidate related form fields when the current form item is validated.

This supports cross-field validation cases such as password confirmation, where changing one field should trigger validation for another dependent field.

Refs https://github.com/element-plus/element-plus/discussions/24298

### Changes

- Add `relations` prop to `FormItem`
- Expose form fields through form context for relation lookup
- Revalidate matched related fields while avoiding recursive relation validation
- Support both string props and array path props
- Add unit tests for relation validation behavior
- Add English docs for the new `relations` API

## Summary by CodeRabbit

* **New Features**
  * Form items now support a relations attribute to declare dependent fields that are automatically revalidated; handles nested/array-style paths, ignores missing/self/null relations, and prevents recursive validation loops.

* **Tests**
  * Added tests covering relations revalidation, matching rules, circular-relation safety, and edge cases.

* **Documentation**
  * FormItem docs updated to describe the new relations attribute.

### CODE REVIEW

File: packages/components/form/src/form-item.vue [hunk 1]
```diff
@@ -301,7 +313,30 @@ const doValidate = async (rules: RuleItem[]): Promise<true> => {
...
+      !relationProps.has(field.propString) ||
+      field.fieldValue == null
+    ) {
+      continue
+    }
+
+    const relationField = field as PrivateFormItemContext
+    relationField.validateRelation('', () => undefined)
+  }
+}
```
[the remark this answers is not in the document]
Comment (guozi9999) [reply packages/components/form/src/form-item.vue:333]: relations is meant to keep dependent fields in sync, not to forward the source field's trigger. If we pass the original trigger, the related field's rules may be filtered out again, so cases like password validating on change while confirmPassword validates on blur would be skipped and leave stale validation state. Using an empty trigger here is intentional so the related field always performs a full revalidation when the dependency changes.

**Comment by guozi9999** [comment]
I tried to add a StackBlitz demo, but the `pkg.pr.new` preview package for this PR does not seem to be available yet. `https://pkg.pr.new/element-plus@24313` currently returns 404, so I cannot make the demo run against the PR build at the moment.

**Comment by guozi9999** [comment]
> [StackBlitz开启](https://pkg.pr.new/template/1ff0abdb-8c3d-4214-95d7-242262c42edb)
> 
> ```
> pnpm add https://pkg.pr.new/element-plus@24313
> ```
> 
> ```
> npm i https://pkg.pr.new/element-plus@24313
> ```
> 
> ```
> yarn add https://pkg.pr.new/element-plus@24313.tgz
> ```
> 
> _承诺： [](https://github.com/element-plus/element-plus/runs/77653719374)_

I added a StackBlitz demo here: https://stackblitz.com/edit/nhdidhmk?file=src%2FApp.vue
﻿

**Comment by guozi9999** [comment]
Hi! Just a friendly ping 🙂 This PR has been open for a few days. Would appreciate it if a maintainer could take a look when convenient. Thanks!

**Comment by guozi9999** [comment]
Hi maintainers, pinging again. This feature (FormItem relations validation) has been used in production at our company for several years and I believe it would benefit the broader Element Plus community.

I would really appreciate your feedback - could you let me know if there is an issue with the PR itself, or if the approach needs to be reconsidered? I am happy to adjust the implementation based on your guidance.

Thank you for your time!

File: packages/components/form/src/form-item.vue [hunk 2]
```diff
@@ -301,7 +313,30 @@ const doValidate = async (rules: RuleItem[]): Promise<true> => {
...
+const validateRelations = () => {
+  if (!formContext || props.relations.length === 0) return
+
+  const relationProps = new Set(props.relations)
+  for (const field of formContext.fields) {
+    if (
+      !field.propString ||
+      field.propString === propString.value ||
+      !relationProps.has(field.propString) ||
+      field.fieldValue == null
```
[the remark this answers is not in the document]
Comment (guozi9999) [reply packages/components/form/src/form-item.vue:325]: This should not happen, because when the corresponding form field changes its own value, it will also trigger its own validation. So its validation state will be updated automatically without needing an extra manual trigger.

**Comment by guozi9999** [comment]
Hi @btea, sorry to bother you.

This PR is still marked as `Needs Review`. Since you recently helped merge several form-related PRs, would you mind taking a look when you have time, or letting me know if there is another maintainer I should ask?

I also saw the latest automated review feedback about relation validation behavior, and I’m happy to adjust those details. Before polishing further, I’d really appreciate some maintainer guidance on whether the proposed `relations` API direction is acceptable.

Thanks!

**Comment by btea** [comment]
Thank you for your contribution.

To be honest, in my opinion, this feature leans more towards a business logic approach. Furthermore, if they can be managed together, nested form options and validation results might also be managed interchangeably, leading to increased complexity.

I prefer that the developers manage this feature themselves within their projects. We can look at other people's opinions. cc @tolking @keeplearning66 @Dsaquel @rzzf

**Comment by rzzf** [comment]
The reason I’m okay with this feature is that I think it can simplify some validation logic. That said, this responsibility is more appropriately handled by wrapper components. Whether this proposal is ultimately accepted or not, I don't have any objections to either outcome.
